api/src/controllers/BatchDeleteController.js

```diff
@@ -0,0 +1,53 @@
+import getAuthFromRequest from 'lib/helpers/getAuthFromRequest';
+import catchErrors from 'api/controllers/utils/catchErrors';
+import getScopeFilter from 'lib/services/auth/filters/getScopeFilter';
+import BatchDelete from 'lib/models/batchDelete';
+import Statement from 'lib/models/statement';
+import getOrgFromAuthInfo from 'lib/services/auth/authInfoSelectors/getOrgFromAuthInfo';
+import parseQuery from 'lib/helpers/parseQuery';
+import { BATCH_STATEMENT_DELETION_QUEUE } from 'lib/constants/batchDelete';
+import { publish } from 'lib/services/queue';
+import { get } from 'lodash';
+
+
+const initialiseBatchDelete = catchErrors(async (req, res) => {
+  // Authenticate
+  const authInfo = getAuthFromRequest(req);
+
+  const scopeFilter = await getScopeFilter({
+    modelName: 'statement',
+    actionName: 'delete',
+    authInfo
+  });
+
+  const filter = {
+    ...(await parseQuery(req.query.filter)),
+    ...scopeFilter
+  };
+
+  const total = await Statement
+    .find(filter)
+    .count();
+
+  console.log('authInfo', JSON.stringify(authInfo, null, 2));
+  const batchDelete = new BatchDelete({
+    organisation: getOrgFromAuthInfo(authInfo),
+    total,
+    filter: req.query.filter,
+    client: get(authInfo, 'client.id')
+  });
+  await batchDelete.save();
+
+  await publish({
```

Comment:
I'm not sure what this publishes to a worker, can you clarify please @cbishopvelti?

Reply:
Chris said that the scheduler will trigger a worker job to delete a batch and the worker job will delete a batch then spawn a subsequent job for the next batch deletion if more deletions are required and we're within the time window for allowing the deletion. Note that the locking in this PR is to ensure the scheduler doesn't run on multiple servers, that locking also exists in other schedulers for the same reason.

Reply:
I think this makes sense - we don't want one process to handle the deletion solely as it would mean the job would be lost if the server fell over etc. 

The locking however doesn't feel right, as in flight queues are only pulled by one worker at a time, so that should be fine. 

Instead, where locking _is_ needed is when the scheduler locks using _Redis_ to ensure no two schedulers can run the scheduled job at the same time. See here: https://github.com/LearningLocker/learninglocker/blob/master/cli/src/scheduler.js#L42

Reply:
Yeah locking is only for the scheduler here.